### PR TITLE
[VPD-1313] PrimeV2 deployments for BSC

### PR DESCRIPTION
**Pull request by Debugger022**
## Summary
- Introduce `PrimeLens` — a standalone view-only contract that reads PrimeV2's public mappings and reconstructs the V1 APR helpers (`calculateAPR`, `estimateAPR`, `incomeDistributionYearly`) for off-chain consumers. Lives in a separate deployment so PrimeV2 stays under the 24,576-byte EVM contract-size limit.
- Restore impl-only / lens-only / leaderboard-only deploy scripts so the three contracts can be redeployed independently across upgrades and chains.
- Land deployment artifacts for bsctestnet + bscmainnet (PrimeV2, PrimeLeaderboard, PrimeLens) plus a fresh testnet WBNB reward-distribution fork test.

### Contracts
- `PrimeLens.sol` — new stateless view contract. Reads PrimeV2's public getters via `IPrimeV2WithStorage`; computes per-(market, user) APR figures off-chain. PrimeV2 itself unchanged → no link-time dependency, no contract-size pressure.
- `IPrimeV2.sol` — added `APRInfo` and `Capital` structs consumed by the lens.

### Deploy scripts
- `deploy/014-deploy-prime-leaderboard.ts` — split out from the original combined script.
- `deploy/014-deploy-prime-v2.ts` — focused on PrimeV2 proxy + initialize, with ACM-permission grants for the local hardhat network and ownership transfer to the network's `NormalTimelock` on live chains.
- `deploy/015-deploy-prime-v2-impl.ts` — impl-only redeploy path (no proxy / initialize); used when shipping a new PrimeV2 implementation for upgrade.
- `deploy/016-deploy-prime-lens.ts` — deploys the lens with the existing PrimeV2 proxy address as constructor arg.
- Removed `deploy/015-configure-prime-v2.ts` — wiring now lives inside the per-contract scripts; on live networks the cross-contract setters run via VIP, not deploy.

### Deployments
- bsctestnet: PrimeV2 (`0xeC22366d2572e52BCB29B50C905b945BA421B9b2`), PrimeLeaderboard, PrimeLens, with implementations + proxies.
- bscmainnet: PrimeV2 (`0x4f5fd115Df31CC48De880a988D74aaD931851628`), PrimeLeaderboard, PrimeLens (`0x43e768EFa1bC365bfd0965A490a41d87c6Dea526`), with implementations + proxies + solcInputs.

### Tests
- `tests/hardhat/PrimeV2/PrimeLens.ts` — unit suite for the lens. Constructor edge cases; exact-equality assertions against PrimeV2 stored state; APR formula identity check; zero-distribution and no-totalScore branches; estimate's "subtract user score then re-add" invariant.
- `tests/hardhat/Fork/PrimeLensForkTest.ts` — bsctestnet fork test. Deploys fresh PrimeV2 + Leaderboard + Lens against real Comptroller/Oracle/XVSVault/PLP/ACM; verifies all three view helpers behave correctly against on-chain state.
- `tests/hardhat/Fork/PrimeV2TestnetWbnbRewardForkTest.ts` — fork test covering the native-market (WBNB) reward path on bsctestnet.

### Config
- `hardhat.config.ts` — added chain 97 (bsctestnet) hardfork history to `isFork()` so testnet fork tests can execute historical blocks.

### CODE REVIEW

File: contracts/Tokens/Prime/PrimeLens.sol [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+        if (totalCappedValue == 0) return (0, 0);
+
+        uint256 userSupplyIncomeYearly = (userYearlyIncome * totalCappedSupply) / totalCappedValue;
+        uint256 userBorrowIncomeYearly = (userYearlyIncome * totalCappedBorrow) / totalCappedValue;
+
+        supplyAPR = totalSupply == 0 ? 0 : ((userSupplyIncomeYearly * MAXIMUM_BPS) / totalSupply);
+        borrowAPR = totalBorrow == 0 ? 0 : ((userBorrowIncomeYearly * MAXIMUM_BPS) / totalBorrow);
+    }
+
+    function _getUnderlying(address vToken) internal view returns (address) {
```
Comment (zed-venus) [inline contracts/Tokens/Prime/PrimeLens.sol:241]: we need handle the native token case, same with prime v2

function _getUnderlying(address vToken) internal view returns (address) {
    if (vToken == primeV2.NATIVE_MARKET()) {
        return primeV2.WRAPPED_NATIVE_TOKEN();
    }
    return IVToken(vToken).underlying();
}

File: contracts/Tokens/Prime/PrimeLens.sol [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+            uint256 rewardIndex,
+            uint256 sumOfMembersScore,
+            bool exists
+        );
+
+    function interests(
+        address market,
+        address user
+    ) external view returns (uint256 accrued, uint256 score, uint256 rewardIndex, uint256 lifetimeAccrued);
+
```
Comment (zed-venus) [inline contracts/Tokens/Prime/PrimeLens.sol:49]: function NATIVE_MARKET() external view returns (address);
function WRAPPED_NATIVE_TOKEN() external view returns (address);

add this two to read from Prime v2

Comment (Debugger022) [reply contracts/Tokens/Prime/PrimeLens.sol:49]: [284eb8c](https://github.com/VenusProtocol/venus-protocol/pull/677/commits/284eb8c51da07cd5dcae171d5f20930e441aea04)

File: contracts/Tokens/Prime/PrimeLens.sol (hunk 1 above)
Comment (Debugger022) [reply contracts/Tokens/Prime/PrimeLens.sol:241]: [284eb8c](https://github.com/VenusProtocol/venus-protocol/pull/677/commits/284eb8c51da07cd5dcae171d5f20930e441aea04)

**Comment by fred-venus** [comment]
merge into original pr so its easier to walk through
